### PR TITLE
block_lvm_read_only: Test of lvm with read-only

### PR DESCRIPTION
**Pull request by leidwang**
block_lvm_read_only:test of lvm with read-only

Signed-off-by: Leidong Wang <leidwang@redhat.com>

ID:1909932

### CODE REVIEW

**Comment by leidwang** [comment]
JOB ID     : a1391782cc71a25a69e6ccce6bbd47fc46f0944c
JOB LOG    : /root/avocado/job-results/job-2020-12-24T00.28-a139178/job.log
 (1/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.lvm_with_read_only: PASS (81.46 s)
 (2/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.lvm_with_read_only.q35: PASS (82.84 s)

**Comment by leidwang** [comment]
@qingwangrh Could you please review this patch?Many thanks!

**Comment by qingwangrh** [comment]
@YongxueHong Please help to review ,thanks.

File: qemu/tests/lvm_with_read_only.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    :param env: Dictionary with test environment
+    """
+
+    def create_lvm():
+        devname = process.getoutput(params["get_devname_command"])
+        pvcreate_command = (params["pvcreate_command"] % devname)
+        process.system(pvcreate_command)
+        vgcreate_command = (params["vgcreate_command"] % devname)
+        process.system(vgcreate_command)
+        process.system(params["lvcreate_command"])
```
Comment (YongxueHong) [inline qemu/tests/lvm_with_read_only.py:27]: Could we call the avocado-vt framework interface `./virttest/lvm.py` to implement this part?

File: qemu/tests/lvm_with_read_only.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    vm = env.get_vm(params['main_vm'])
+    session = vm.wait_for_login()
+    qmp_port = vm.monitor
+
+    try:
+        qmp_port.send_args_cmd(params["unplug_qmp_command0"])
+        time.sleep(5)
+        qmp_port.send_args_cmd(params["unplug_qmp_command1"])
+        qmp_port.send_args_cmd(params["unplug_qmp_command2"])
+        qmp_port.send_args_cmd(params["plug_qmp_command"])
```
Comment (YongxueHong) [inline qemu/tests/lvm_with_read_only.py:41]: Please use `simple_unplug` from `./virttest/qemu_devices/qcontainer.py` or you could use `BlockDevicesPlug.unplug_devs_serial` from `./test-providers.d/downloads/io-github-autotest-qemu/provider/block_devices_plug.py`

**Comment by leidwang** [comment]
depend on [https://github.com/avocado-framework/avocado-vt/pull/2880](url)

File: qemu/tests/lvm_with_read_only.py (hunk 2 above)
Comment (leidwang) [reply qemu/tests/lvm_with_read_only.py]: Thanks for your guidance and updated the code. Could you please review this patch again?

**Comment by leidwang** [comment]
JOB ID     : 5228b62d8f76e940fd5429ba06cba057e821a79e
JOB LOG    : /root/avocado/job-results/job-2021-01-08T02.34-5228b62/job.log
 (1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.lvm_with_read_only.q35: PASS (73.26 s)

**Comment by leidwang** [comment]
JOB ID     : 9501cdcc022f0a021cbb460b30d550a460c6e023
JOB LOG    : /root/avocado/job-results/job-2021-01-11T04.02-9501cdc/job.log
 (1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.block_lvm_read_only.q35: PASS (66.09 s)

File: qemu/tests/cfg/block_lvm_read_only.cfg [hunk 3]
```diff
@@ -0,0 +1,27 @@
+- block_lvm_read_only:
+    virt_test_type = qemu
+    type = block_lvm_read_only
+    only virtio_scsi
+    start_vm = no
+    monitors = 'qmp1'
+    monitor_type_qmp1 = qmp
```
[the remark this answers is not in the document]
Comment (leidwang) [reply qemu/tests/cfg/block_lvm_read_only.cfg]: Yes, what you said makes sense and updated the code based on your suggestions.Thanks a lot.

**Review by YongxueHong — APPROVED** [review]

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=block_lvm_read_only --guestname=RHEL.8.4.0 --driveformat=virtio_scsi,virtio_blk --clone=no

(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.block_lvm_read_only: PASS (50.27 s)
LGTM

**Comment by YongxueHong** [comment]
Please update your commit header to `block_lvm_read_only: Test of lvm with read-only`

**Comment by leidwang** [comment]
> Please update your commit header to `block_lvm_read_only: Test of lvm with read-only`

Sorry, will update it.

**Pull request merged**